addons/delivery/tests/test_delivery_cost.py

```diff
@@ -155,3 +155,71 @@ def test_00_delivery_cost(self):
         self.default_delivery_policy = self.SaleConfigSetting.create({})
 
         self.default_delivery_policy.execute()
+
+    def test_01_delivery_cost_from_pricelist(self):
+        """ This test aims to validate the use of a pricelist to compute the delivery cost in the case the associated
+            product of the shipping method is defined in the pricelist """
+
+        #PART 1: CREATION OF A SO WITH STANDARD SHIPPING METHOD AND PUBLIC PRICE
```

Comment:
The basic flow seems already checked by the other test
I think you can just test your issue

Reply:
The idea of adding the basic case was to make a comparison between a SO without the pricelist (ie to check we use the standard price) and a SO with the pricelist (to check we use our price), but you're totally right, it's already tested in the same way.
Do you think it should be useful to let this part if I move the definition of the pricelist at the beginning of the test (so that our custom price will be defined before the first SO and we will make sure we don't use the custom price) or it's still useless ?